src/main/java/co/cask/gcp/gcs/sink/GCSMultiBatchSink.java

```diff
@@ -0,0 +1,191 @@
+    super.initialize(context);
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    config.validate();
+    super.configurePipeline(pipelineConfigurer);
+  }
+
+  @Override
+  public void prepareRun(BatchSinkContext context) throws IOException, UnsupportedTypeException {
+    config.validate();
+    for (Map.Entry<String, String> argument : context.getArguments()) {
+      String key = argument.getKey();
+      if (!key.startsWith(TABLE_PREFIX)) {
+        continue;
+      }
+      String name = key.substring(TABLE_PREFIX.length());
+
+      String schema = argument.getValue();
+
+      Job job = JobUtils.createInstance();
+      Configuration outputConfig = job.getConfiguration();
+
+      outputConfig.set(FileOutputFormat.OUTDIR, config.getOutputDir(context.getLogicalStartTime(), name));
+      Map<String, String> gcsProperties = GCPUtils.getFileSystemProperties(config);
+      for (Map.Entry<String, String> entry : gcsProperties.entrySet()) {
+        outputConfig.set(entry.getKey(), entry.getValue());
+      }
+
+
+      outputConfig.set(RecordFilterOutputFormat.PASS_VALUE, name);
+      outputConfig.set(RecordFilterOutputFormat.ORIGINAL_SCHEMA, schema);
+      String format = config.getOutputFormat();
+      outputConfig.set(RecordFilterOutputFormat.FORMAT, format);
+
+
+      outputConfig.set(RecordFilterOutputFormat.FILTER_FIELD, config.splitField);
+      job.setOutputValueClass(org.apache.hadoop.io.NullWritable.class);
+      if (RecordFilterOutputFormat.AVRO.equals(format)) {
+        org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(schema);
+        Map<String, String> configuration =
+                FileSetUtil.getAvroCompressionConfiguration(config.codec, schema, false);
+        for (Map.Entry<String, String> entry : configuration.entrySet()) {
+          outputConfig.set(entry.getKey(), entry.getValue());
+        }
+        AvroJob.setOutputKeySchema(job, avroSchema);
+      } else if (RecordFilterOutputFormat.ORC.equals(format)) {
+        StringBuilder builder = new StringBuilder();
+        co.cask.hydrator.common.HiveSchemaConverter.appendType(builder, Schema.parseJson(schema));
+        outputConfig.set("orc.mapred.output.schema", builder.toString());
+      } else if (RecordFilterOutputFormat.PARQUET.equals(format)) {
+        Map<String, String> configuration =
+                FileSetUtil.getParquetCompressionConfiguration(config.codec, schema, false);
+        for (Map.Entry<String, String> entry : configuration.entrySet()) {
+          outputConfig.set(entry.getKey(), entry.getValue());
+        }
+      } else {
+        // Encode the delimiter to base64 to support control characters. Otherwise serializing it in Cconf would result
+        // in an error
+        outputConfig.set(RecordFilterOutputFormat.DELIMITER, config.getDelimiter());
+      }
+
+
+      context.addOutput(Output.of(config.getReferenceName(),
```

Comment:
does this work? I would expect it to add outputs of the same name and overwrite each other. Seems like it should be using the name from the argument and not the reference name

Reply:
reference name is also a funny thing to have for this plugin, as it's conceptually writing to many different datasets. I don't think it makes sense to have a reference name, but we could have a reference prefix, where the name from the argument is appended to that prefix.

Reply:
changed to use name instead of reference name.
Both did not changed the functionality, it works with reference name and with name

Reply:
Hm ok CDAP must be generating some id for it somewhere then.

The name used here will be the one that is used for tracking lineage. If it's just the name, it can generate misleading lineage graphs if there are db tables that have the same name, but live in different databases. Let's just change it back to the reference name.